### PR TITLE
Closes SI-6952: add correct error positions for Dynamic feature check.

### PR DESCRIPTION
**Pull request by namin**
Review by @adriaanm.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala [hunk 1]
```diff
@@ -1740,8 +1740,8 @@ trait Typers extends Modes with Adaptations with Tags {
      */
     def validateParentClasses(parents: List[Tree], selfType: Type) {
       val pending = ListBuffer[AbsTypeError]()
-      def validateDynamicParent(parent: Symbol) =
-        if (parent == DynamicClass) checkFeature(parent.pos, DynamicsFeature)
+      def validateDynamicParent(parent: Symbol, parentPos: Position) =
+        if (parent == DynamicClass) checkFeature(parentPos, DynamicsFeature)
```
Comment (xeno-by) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala:1744]: why is `DynamicClass.pos` equal to NoPosition here?

Comment (namin) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala:1744]: Well, DynamicClass is in the standard library. We're just compiling a file, and want errors reported in that file.

Comment (xeno-by) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala:1744]: Sure, I was just wondering why DynamicClass has empty position, but then I remembered that we don't pickle positions.

**Comment by xeno-by** [comment]
lgtm

**Comment by adriaanm** [comment]
LGTM

that was fast -- thanks!

**Pull request merged**